server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -35,6 +39,7 @@ public class DiscoveryExtensionNode extends DiscoveryNode implements Writeable,
     private Version minimumCompatibleVersion;
     private List<ExtensionDependency> dependencies = Collections.emptyList();
     private List<String> implementedInterfaces = Collections.emptyList();
+    private List<Scope> scopes = List.of();
 
     public DiscoveryExtensionNode(
```

Comment:
Rather than duplicate the scopes can we instead reference the settings (maybe via the extension manager)?